elegy/losses/mean_absolute_percentage_error.py

```diff
@@ -0,0 +1,142 @@
+
+    y_true = jax.random.randint(rng, shape=(2, 3), minval=0, maxval=2)
+    y_pred = jax.random.uniform(rng, shape=(2, 3))
+
+    loss = elegy.losses.mean_percentage_absolute_error(y_true, y_pred)
+
+    assert loss.shape == (2,)
+
+    assert jnp.array_equal(loss, 100. * jnp.mean(jnp.abs((y_pred - y_true) / jnp.clip(y_true, utils.EPSILON, None))))
+    ```
+    
+    Arguments:
+        y_true: Ground truth values. shape = `[batch_size, d0, .. dN]`.
+        y_pred: The predicted values. shape = `[batch_size, d0, .. dN]`.
+
+    Returns:
+        Mean absolute percentage error values. shape = `[batch_size, d0, .. dN-1]`.
+    """
+    y_true = y_true.astype(y_pred.dtype)
+    diff = jnp.abs((y_pred - y_true) / jnp.clip(y_true, utils.EPSILON, None))
+    return 100. * jnp.mean(diff, axis=-1)
+
+
+class MeanAbsolutePercentageError(Loss):
+    """
+    Computes the mean absolute errors between labels and predictions.
+
+    `loss = mean(abs((y_true - y_pred) / y_true))`
+
+    Usage:
+
+    ```python
+    y_true = jnp.array([[1.0, 1.0], [0.9, 0.0]])
+    y_pred = jnp.array([[1.0, 1.0], [1.0, 0.0]])
+
+    # Using 'auto'/'sum_over_batch_size' reduction type.
+    mape = elegy.losses.MeanAbsolutePercentageError()
+    result = mape(y_true, y_pred)
+    assert jnp.isclose(result, 2.78, rtol=0.01)
+
+    # Calling with 'sample_weight'.
+    assert jnp.isclose(mape(y_true, y_pred, sample_weight=jnp.array([0.1, 0.9])), 2.5, rtol=0.01)
+
+    # Using 'sum' reduction type.
+    mape = elegy.losses.MeanAbsolutePercentageError(reduction=elegy.losses.Reduction.SUM)
+
+    assert jnp.isclose(mape(y_true, y_pred), 5.6, rtol=0.01)
+
+    # Using 'none' reduction type.
+    mape = elegy.losses.MeanAbsolutePercentageError(reduction=elegy.losses.Reduction.NONE)
+
+    assert jnp.all(jnp.isclose(result, [0. , 5.6], rtol=0.01))
+
+    ```
+    Usage with the Elegy API:
+
+    ```python
+    model = elegy.Model(
+        module_fn,
+        loss=elegy.losses.MeanAbsolutePercentageError(),
+        metrics=elegy.metrics.Mean.defer(),
+    )
+    ```
+    """
+
+    def __init__(
+        self,
+        reduction: tp.Optional[Reduction] = None,
+        name: tp.Optional[str] = None,
+        weight: tp.Optional[float] = None,
+        on: tp.Optional[types.IndexLike] = None,
+    ):
+        """
+        Initializes `Mean` class.
+
+        Arguments:
+            reduction: (Optional) Type of `elegy.losses.Reduction` to apply to
+                loss. Default value is `SUM_OVER_BATCH_SIZE`. For almost all cases
+                this defaults to `SUM_OVER_BATCH_SIZE`. When used with
+                `tf.distribute.Strategy`, outside of built-in training loops such as
+                `elegy` `compile` and `fit`, or `SUM_OVER_BATCH_SIZE`
+                will raise an error.
+                for more details.
+            name: Optional name for the loss.
+            weight: Optional weight contribution for the total loss. Defaults to `1`.
+        """
```

Comment:
Please add documentation for the `on` argument, use the same as in other metrics.